.mvn/modernizer/violations.xml

```diff
@@ -158,6 +158,12 @@
         <comment>Prefer ConfigurationInstantiator.newEmptyConfiguration() for two reasons: (1) loading default resources is unlikely desired and (2) ConfigurationInstantiator adds additional safety checks</comment>
     </violation>
 
+    <violation>
+        <name>org/apache/hadoop/fs/FileSystem.close:()V</name>
+        <version>1.1</version>
+        <comment>Hadoop FileSystem instances are singletons and should not be closed</comment>
```

Comment:
"singletons" or "shared"?

Reply:
Not sure I understand the question. They are singletons and thus are shared.

Reply:
Would it be more clear if I wrote "shared singletons"?

Reply:
I don't like plural "singletons". To me a singletons is a one object.
Pool of shared objects is "shared objects" to me.

Also, FS instances have some lifecycle, right? They are created, cached and closed behind the scenes, am i correct?
So the application can eventually create unlimited number of these objects over time.

Reply:
There's no contradiction in the plural form of singletons. Each type of FileSystem can be a singleton (one for S3, one for HDFS, one for Minio, etc), and you can still refer to them collectively as a group of singletons.

Reply:
So we create as many FileSystem objects as many different URI schemes are in use?
Does the answer depend on extra credentials like for GCS?

Reply:
I think it's best to say that "it's complicated" and you can find the exact logic is in `TrinoFileSystemCache`. I changed the wording to "shared" since it seems to be a better way to communicate why we shouldn't close them.